src/core/moves/MetropolisHastingsMove.cpp

```diff
@@ -341,8 +344,6 @@ void MetropolisHastingsMove::performMcmcMove( double prHeat, double lHeat, doubl
     // compute the probability of the current value for each node
     for (auto node: views::concat(touched_nodes, affected_nodes))
     {
-        if (fail_probability) break;
```

Comment:
Removing this line means that we need to deal with probabilities that throw an exception.

I removed this because I thought it was fragile, since it relies on the order of the nodes, which is not fully determined.

Reply:
I suspect that this was necessary to match previous behavior, but I can't quite remember...